apps/gallery/test/marionette/lib/system.js

```diff
@@ -25,6 +26,13 @@ System.prototype = {
     this.client.apps.switchToApp(appOrigin);
   },
 
+  waitForActionMenuToDisplay: function() {
+    this.client.waitFor(function() {
+      return this.client.helper.waitForElement(
+        System.Selector.actionMenu).displayed();
+    }.bind(this));
+  },
+
```

Comment:
I'm surprised that you need this new function. I thought that the point of helper.waitForElement() is that it waited for the element to be displayed.  I assume it is here because it wasn't working without it, but I don't understand how this fixes anything, do you?